src/Microsoft.Graph.Core/Requests/Middleware/AuthenticationHandler.cs

```diff
@@ -149,14 +149,15 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
         /// <param name="newRequest">Request message to add claims to</param>
         private void AddClaimsToRequestContext(HttpRequestMessage newRequest, string wwwAuthenticateHeader)
         {
-            int claimsStart = wwwAuthenticateHeader.IndexOf("claims=", StringComparison.OrdinalIgnoreCase);
-            if (claimsStart < 0) 
+            var claimsHeaderString = wwwAuthenticateHeader.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault( header => header.Contains("claims="));
+            if (string.IsNullOrEmpty(claimsHeaderString)) 
                 return; // do nothing as there is no claims in www Authenticate Header
 
-            claimsStart += 8; // jump to the index after the opening quotation mark
+            claimsHeaderString = claimsHeaderString.Trim();// remove any potential whitespace
+            int claimsStart = 8; // jump to the index after the opening quotation mark
```

Comment:
Are there scenarios where this index might be different?

Reply:
Not really. The first 8 characters will be `claims="`

Reply:
```c#
Regex ClaimsRegex = new Regex(@"(claims="")([\w=]+)("")", RegexOptions.Singleline | RegexOptions.Compiled);
var match = ClaimsRegex.Match(wwwAuthenticateHeader);
if(match.Success){
    var claimsHeaderString = match.Groups[2].Value;
}

```

Reply:
Thanks for this @rkodev. 
Unfortunately, I had set the PR to auto-merge. So, I have highlighted the feedback in #462